### PR TITLE
Add Developer Guide details for Architecture, Summary Features and Tab Switching features

### LINKED ISSUE #368: Add DG Details for Summary Feature, Tab Switching and Architecture details

### PR DESCRIPTION
**Pull request by ngzhenteng**
Fixes #368

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -54,13 +55,13 @@ The ***Architecture Diagram*** given above explains the high-level design of the
 
 [**`Commons`**](#common-classes) represents a collection of classes used by multiple other components.
 
-The rest of the App consists of five components.
+The rest of Jarvis consists of five components.
```
Comment (peironggg) [inline docs/DeveloperGuide.md:59]: Great attention to detail!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -177,6 +184,81 @@ Classes used by multiple components are in the `seedu.jarvis.commons` package.
 
 This section describes some noteworthy details on how certain key features have been implemented.
 
+### Automatic Tab Switching
```
Comment (peironggg) [inline docs/DeveloperGuide.md:187]: Real comprehensive explanation 👍

File: docs/diagrams/SummaryFeatureSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+@startuml
```
Comment (peironggg) [inline docs/diagrams/SummaryFeatureSequenceDiagram.puml:1]: nice pummel diagrams

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Asuraxsoul) [reply docs/DeveloperGuide.md:187]: 👍

File: docs/diagrams/UiClassDiagram.puml [hunk 4]
```diff
@@ -11,8 +11,8 @@ Class UiManager
 Class MainWindow
 Class HelpWindow
 Class ResultDisplay
-Class PersonListPanel
-Class PersonCard
+Class XYZListPanel
```
Comment (wilinetan) [inline docs/diagrams/UiClassDiagram.puml:14]: thanks for refactoring the diagrams

**Review by wilinetan — COMMENTED** [review]
LGTM

File: docs/diagrams/SummaryFeatureSequenceDiagram.puml (hunk 3 above)
Comment (ngzhenteng) [reply docs/diagrams/SummaryFeatureSequenceDiagram.puml:1]: Thank you!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (wilinetan) [reply docs/DeveloperGuide.md:187]: 👍

**Comment by Asuraxsoul** [comment]
LGTM

**Pull request merged**